From 20472f229151ca848196a1d3276d9685cad662a0 Mon Sep 17 00:00:00 2001
From: Luca Matei Pintilie <luca@lucamatei.com>
Date: Thu, 6 Jul 2023 21:02:59 +0200
Subject: [PATCH] SDK telemetry optout

Disables telemetry by default.

Patch copied from Alpine Linux
and authored by Antoine Martin <dev@ayakael.net>
---
 src/sdk/src/Cli/dotnet/Program.cs | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/sdk/src/Cli/dotnet/Program.cs b/src/sdk/src/Cli/dotnet/Program.cs
index 32a9f4ccc6..ac3f5fd293 100644
--- a/src/sdk/src/Cli/dotnet/Program.cs
+++ b/src/sdk/src/Cli/dotnet/Program.cs
@@ -27,6 +27,12 @@ public class Program
 
         public static int Main(string[] args)
         {
+            // opt out of telemetry by default if the env var is unset
+            string telemetryValue = Environment.GetEnvironmentVariable("DOTNET_CLI_TELEMETRY_OPTOUT");
+            if (String.IsNullOrEmpty(telemetryValue))
+            {
+                Environment.SetEnvironmentVariable("DOTNET_CLI_TELEMETRY_OPTOUT", "1");
+            }
             using AutomaticEncodingRestorer _ = new();
 
             // Setting output encoding is not available on those platforms
-- 
2.41.0

